File: cs/sample/Sample.Tests/Merge2SortedLs/testMerge2SortedLs.cs
using System; 
using Sample.Merge2SortedLs;
using Xunit;
using Sample.ListNode;

namespace Sample.Tests{
    public class UnitTest_Merge2SortedLs{
        [Theory]
        [InlineData("124","134","112344")]
        [InlineData("","","")]
        [InlineData("1","","1")]
        public void checkResult(string n1,string n2,string expected){
            listNode l1=merge2SortedLs.listNode_fromStr(n1);
            listNode l2=merge2SortedLs.listNode_fromStr(n2);
            listNode l3=merge2SortedLs.recursive(l1,l2);
            string n3=merge2SortedLs.printListNode(l3);
            Assert.Equal(expected,n3);
        }
    }

}

File: py/tests/testMerge2SortedLs/test_Merge2SortedLs.py
import unittest
from parameterized import parameterized as p
from solns.merge2SortedLs.merge2SortedLs import *

class UnitTest_Merge2SortedLs(unittest.TestCase):
    @p.expand([
        ["124","134","112344"],
        ["","",""],
        ["1","","1"]
    ])
    def test_recursive(self,n1,n2,expected):
        l1 = Solution.listNode_fromStr(n1)
        l2 = Solution.listNode_fromStr(n2)
        l3 = Solution.recursive(l1,l2)
        n3 = Solution.printListNode(l3)
        self.assertEqual(n3,expected)